Use a matrix to check and push to multiple charms

rbarry82 opened the pull request:
## Issue
Expand the github workflows so they automatically publish resources to any charm which needs it.

## Solution
Use a matrix in the workflow instead of hardcoding prometheus only

## Release Notes
Automatically publish resources to more charms than just prometheus

rbarry82 committed:
Use a matrix to check and push to multiple charms

sed-i commented on .github/workflows/release.yaml:
```diff
@@ -71,7 +74,7 @@ jobs:
         # This assumes that the amd64 and arm64 revisions are in lockstep.
         run: |
-          echo "::set-output name=rev::$(($(charmcraft resource-revisions prometheus-k8s promql-transform-amd64 | tail -n+2 | head -1 | awk '{ print $1 }') + 1))"
+          echo "::set-output name=rev::$(($(charmcraft resource-revisions ${{ matrix.charm }} promql-transform-amd64 | tail -n+2 | head -1 | awk '{ print $1 }') + 1))"
```
Iiuc, `rev` is used here for tagging releases in _this_ repo, however:
1. A given charm can have different resource revisions for amd64 and arm64, but sticking with just one of them is fine, except -
2. prometheus-k8s and grafana-k8s and future additions would all have different resource revision numbers for the same resource. Would multiple `rev` tags be created?
3. A race condition between this and manual cli is technically possible. Can we instead parse the output of the `upload-resource` command?

mmanciop replied:
(1) is true, we should use a matrix for the arch too and that would solve itself.
(2) creating multiple rev tags is fine, although not optimal, as long as the tag has the arch in the tag name.
(3) Problem not worth solving IMO.

sed-i commented on .github/workflows/release.yaml:
```diff
@@ -116,9 +119,9 @@ jobs:
           CHARMCRAFT_AUTH: ${{ secrets.CHARMHUB_TOKEN }}
         run: |
-          charmcraft upload-resource prometheus-k8s promql-transform-amd64 --file artifacts/promql-transform-amd64
+          charmcraft upload-resource ${{ matrix.charm }} promql-transform-amd64 --file artifacts/promql-transform-amd64
```
How does this action behave if `charmcraft resources` doesn't include both `promql-transform-amd64` and `promql-transform-arm64` for the given charm?

rbarry82 replied:
This will soon be irrelevant, since they'll all get the same `charmcraft-parts` change as Prometheus

mmanciop reviewed (changes requested):
Leon's comment deserve addressing, I'll push a change

Commit:
Overhaul release process for multiple charms

This change reworks the release process for promql-transform to
work with multiple charms and to have separate revisions for
different architectures and charms.

The changes are as follows:
1. The Git tag for the GitHub release is now based on the release day
2. Additional Git tags are created for each charm and architecture for
   the sake of traceability based on the actual revision from CharmHub,
   which is obtained by parsing the output of charmcraft upload-resource
